### PR TITLE
[WT-347] add confirmation dialog when changing plan

### PR DESCRIPTION
**Pull request by rafijv**
When the user changes plan, show a confirmation dialog that explains what the user is being charged and the change in storage capacity.

### CODE REVIEW

File: src/app/core/views/Preferences/tabs/Plans/PlanSelector.tsx [hunk 1]
```diff
@@ -206,3 +226,130 @@ function Price({
...
+  return (
+    <Modal isOpen={isDialgOpen} onClose={onClose}>
+      <h3 className="mb-5 text-2xl font-medium">{translate('views.account.tabs.plans.dialog.title')}</h3>
+      <p className="font-regular mb-9 text-lg">
+        {translate('views.account.tabs.plans.dialog.subtitle1')}
+        <span className="font-semibold">{currentPlanSizeString}</span>
+        {translate('views.account.tabs.plans.dialog.subtitle2')}
+        <span className="font-semibold">{selectedPlanSizeString}</span>
+        {translate('views.account.tabs.plans.dialog.subtitle3')}
+      </p>
```
Comment (CandelR) [inline src/app/core/views/Preferences/tabs/Plans/PlanSelector.tsx:288]: Why didn't you pass the variables to the i18n translations? This accesses the whole sentence with a key and keeps all the context for generating translations.

File: src/app/core/views/Preferences/tabs/Plans/PlanSelector.tsx [hunk 2]
```diff
@@ -206,3 +226,130 @@ function Price({
...
+      {selectedPlanSize < planUsage && (
+        <div className="mb-5 flex flex-col items-center rounded-xl border border-red-30 bg-red-10 px-4 py-5 text-red-std">
+          <h4 className="mb-1.5 text-center text-xl font-semibold">
+            {translate('views.account.tabs.plans.dialog.alert.title')}
+            {selectedPlanSizeString}
+          </h4>
+          <p className="font-regular text-center text-base">
+            {translate('views.account.tabs.plans.dialog.alert.text1')}
+            <span className="font-semibold">{bytesToString(planUsage)}</span>
+            {translate('views.account.tabs.plans.dialog.alert.text2')}
```
Comment (CandelR) [inline src/app/core/views/Preferences/tabs/Plans/PlanSelector.tsx:338]: Same here with the i18n variables

File: src/app/core/views/Preferences/tabs/Plans/PlanSelector.tsx [hunk 3]
```diff
@@ -206,3 +226,130 @@ function Price({
...
 }
+
+const ChangePlanDialog = ({
+  prices,
+  isDialgOpen,
+  setIsDialogOpen,
+  onPlanClick,
+  priceIdSelected,
+}: {
+  prices;
```
Comment (CandelR) [inline src/app/core/views/Preferences/tabs/Plans/PlanSelector.tsx]: you forgot to add the type of prices

File: src/app/core/views/Preferences/tabs/Plans/PlanSelector.tsx [hunk 4]
```diff
@@ -206,3 +226,130 @@ function Price({
...
+  isDialgOpen: boolean;
+  setIsDialogOpen: (value: boolean) => void;
+  onPlanClick: (value: string) => void;
+  priceIdSelected: string;
+}): JSX.Element => {
+  const plan = useSelector<RootState, PlanState>((state) => state.plan);
+  const { translate } = useTranslationContext();
+
+  const { planLimit, planUsage, subscription } = plan;
+  const selectedPlan = prices.filter((plan) => plan.id === priceIdSelected);
```
Comment (CandelR) [inline src/app/core/views/Preferences/tabs/Plans/PlanSelector.tsx]: If the selected plan can only be used once it is better to use find, there is no need for an array to store a single value.

**Review by CandelR — CHANGES_REQUESTED** [review]

File: src/app/core/views/Preferences/tabs/Plans/PlanSelector.tsx (hunk 1 above)
Comment (rafijv) [reply src/app/core/views/Preferences/tabs/Plans/PlanSelector.tsx:288]: Because I also have to give it css styles and I haven't found the way to do it.

Comment (CandelR) [reply src/app/core/views/Preferences/tabs/Plans/PlanSelector.tsx:288]: Understood! 👌

File: src/app/core/views/Preferences/tabs/Plans/PlanSelector.tsx (hunk 3 above)
Comment (rafijv) [reply src/app/core/views/Preferences/tabs/Plans/PlanSelector.tsx]: done

File: src/app/core/views/Preferences/tabs/Plans/PlanSelector.tsx (hunk 4 above)
Comment (rafijv) [reply src/app/core/views/Preferences/tabs/Plans/PlanSelector.tsx]: done

**Review by CandelR — APPROVED** [review]

**Pull request merged**